pkg/async/pool/timeout.go

```diff
@@ -11,7 +11,14 @@ import (
 	"github.com/getoutreach/gobox/pkg/async"
 )
 
-// WithTimeout creates enqueuer that cancel enqueueing after given timeout
+// WithTimeout creates enqueuer that cancel enqueueing after given
+// timeout
+//
+// Deprecated: This library is being deprecated in favor of using
+// https://pkg.go.dev/github.com/sourcegraph/conc/pool instead. There is
+// no equivalent to this function in the new library. For more
+// information, see the README:
+// https://github.com/getoutreach/gobox/tree/main/pkg/async/pool/README.md
 func WithTimeout(timeout time.Duration, scheduler Scheduler) Scheduler {
```

Comment:
Nothing at Outreach called this besides one of our `argocd` microservices. Didn't seem worth keeping, but also doesn't work in the new "always block if not available model".